core/src/main/scala/sjsonnew/JavaExtraFormats.scala

```diff
@@ -44,7 +44,7 @@ trait JavaExtraFormats {
   implicit val fileStringIso: IsoString[File] = IsoString.iso[File](
     (f: File) => {
       if (f.isAbsolute) {
-        f.toPath.toUri.toASCIIString
+        new URI(FileScheme, normalizeName(f.getAbsolutePath), null).toASCIIString
```

Comment:
How would this encode `C:\Documents and Settings\` on Windows? It's recommended that this should be `file:///C:/Documents%20and%20Settings/` to maximize the compatibility with current and previous RFCs. - https://eed3si9n.com/encoding-file-path-as-URI-reference

Reply:
I was hoping some CI would cover that, but no :( I will try with a windows VM :crossed_fingers:

Reply:
it returns `file:C:/Documents%20and%20Settings/`

![image](https://user-images.githubusercontent.com/943051/100005907-98c64200-2dc1-11eb-9009-dabf5b294a18.png)

Reply:
which then fails with `java.lang.IllegalArgumentException: URI is not hierarchical`

![image](https://user-images.githubusercontent.com/943051/100006849-0e7edd80-2dc3-11eb-966c-35ab799d51f3.png)

(sorry for the pictures instead of text)

Reply:
I pushed some changes and this is the result in windows

```
Type in expressions for evaluation. Or try :help.

scala> import sjsonnew.IsoString
import sjsonnew.IsoString

scala> import sjsonnew.BasicJsonProtocol._
import sjsonnew.BasicJsonProtocol._

scala> val isoFile = implicitly[IsoString[java.io.File]]
isoFile: sjsonnew.IsoString[java.io.File] = sjsonnew.IsoString$$anon$1@1cea930f

scala> isoFile.to(new java.io.File("."))
res0: String = .

scala> isoFile.to(new java.io.File(".").getAbsoluteFile)
res1: String = file:///C:/Users/IEUser/Desktop/meu%20git/sjson-new/.

scala> new java.net.URI(res1)
res2: java.net.URI = file:///C:/Users/IEUser/Desktop/meu%20git/sjson-new/.

scala> new java.io.File(new java.net.URI(res1))
res3: java.io.File = C:\Users\IEUser\Desktop\meu git\sjson-new\.

scala> res3.isDirectory
res4: Boolean = true

scala>
```